src/foam/nanos/approval/ApprovalDAO.js

```diff
@@ -79,6 +83,16 @@ foam.CLASS({
               getDelegate().put(request);
             }
           }
+        } else if ( objToReput instanceof Retriable && ((Retriable) objToReput).getIsRetriable() ) {
+          // reput retriable object
+          try {
+            rePutObject(x, request);
+            getDelegate().put(request);
+          } catch ( Exception e ) {
+            request.setStatus(ApprovalStatus.REQUESTED);
+            getDelegate().put(request);
```

Comment:
if getDelegate.put is throwing exception  it will throw again here

Reply:
removed try catch block. On retry request is already set to requested so no need to reput on exception